collated/main/A0139942W.md

```diff
@@ -0,0 +1,830 @@
+
+        try {	
+
+            parser.parse(args);
+            final boolean checkMark = prepareMarkArgs();
+            final String find_type = prepareFindTypes();
+            if (!isValidArgs(find_type, args.trim())) {
+                return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                        FindCommand.MESSAGE_USAGE));
+            }
+            
+            
+            final  Set<String> keywordSet = prepareSets(find_type);
+            return new FindCommand(keywordSet, find_type, checkMark);
+            
+        } catch (IncorrectCommandException e) {
+             return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                     FindCommand.MESSAGE_USAGE));
+        
+        } catch (IllegalValueException e) {
+			// TODO Auto-generated catch block
+        	return new IncorrectCommand(e.getMessage());
+		}
+        
+        // keywords delimited by whitespace
+
+
+
+    }
+    
+
+
+	private Set<String> prepareSets(String find_type) throws IncorrectCommandException {
+		// TODO Auto-generated method stub
+    	final String[] keywords = getKeywords(find_type);
+    	final Set<String> preparedKeywordSet = Sets.newHashSet(Arrays.asList(keywords));
+    	preparedKeywordSet.remove(EMPTY_STRING);
+		return preparedKeywordSet;
+	}
+
+	private ArgumentsParser prepareParser() {
+		// TODO Auto-generated method stub
+    	ArgumentsParser prepareParser = new ArgumentsParser();
+    	prepareParser
+    	.addOptionalArg(CommandArgs.FIND_NAME)
+    	.addOptionalArg(CommandArgs.FIND_WEEK)
+    	.addOptionalArg(CommandArgs.FIND_DAY)
+    	.addOptionalArg(CommandArgs.FIND_DESC)
+    	.addOptionalArg(CommandArgs.FIND_TAG)
+    	.addOptionalArg(CommandArgs.FIND_MARK);
+    	return prepareParser;
+	}
+
+	//To check that users does not enter anything between find command and search type
+    //eg. find abcd all/KEYWORDS
+    private boolean isValidArgs(String find_type, String args) {
+		// TODO Auto-generated method stub
+    	int compareCharAt ;
+    	for (compareCharAt = 0 ; compareCharAt < find_type.length() ; compareCharAt++) {
+    		char findType_char = find_type.toLowerCase().charAt(compareCharAt);
+    		char args_char = args.toLowerCase().charAt(compareCharAt);
+ 
+    		if (findType_char != args_char) {
+    			return false;
+    		}
+    	}
+    	
+		int seperatorIndex = compareCharAt++;
+		return args.startsWith(SEPERATOR, seperatorIndex);
+	}
+
+	public String prepareFindTypes() throws IncorrectCommandException {
+    	
+		String name = getNameArg();
+		String week = getWeekArg();
+		String day = getDayArg();
+		String desc = getDescArg();
+		String tag = getTagArg();
+		
+		return getFindTypesArgs(name, week, day, desc, tag);
+      	
+    }
+	
+    private boolean prepareMarkArgs() throws IncorrectCommandException {
+		// TODO Auto-generated method stub
+		String markArgs = getMarkArg();
+		if ("true".equalsIgnoreCase(markArgs)) {
```

Comment:
This is not a proper way of using this. Use a String checkVal="true";
 checkVal.equals().Kind of constants should define above in the class